pallets/xyk/src/lib.rs

```diff
@@ -420,7 +442,7 @@ decl_module! {
             let liquidity_asset_id = Self::get_liquidity_asset(first_asset_id, second_asset_id);
 
             ensure!(
-                <assets::Module<T>>::balance(liquidity_asset_id, sender.clone()) >= liquidity_asset_amount,
+                T::Currency::can_slash(liquidity_asset_id, &sender, liquidity_asset_amount),
```

Comment:
I think we also need to check ensure_can_withdraw, as I believe that when burning frozen amount should be considered.